PWGHF/HFC/TableProducer/correlatorLcHadrons.cxx

```diff
@@ -0,0 +1,700 @@
+          registry.fill(HIST("hMassLcMcRecBkg"), hfHelper.invMassLcToPKPi(candidate), candidate.pt(), efficiencyWeight);
+          registry.fill(HIST("hMassLcVsPtMcRec"), hfHelper.invMassLcToPKPi(candidate), candidate.pt(), efficiencyWeight);
+          registry.fill(HIST("hSelectionStatusMcRec"), candidate.isSelLcToPKPi());
+        }
+        if (candidate.isSelLcToPiKP() >= selectionFlagLc) {
+          registry.fill(HIST("hMassLcMcRec"), hfHelper.invMassLcToPiKP(candidate), efficiencyWeight);
+          registry.fill(HIST("hMassLcMcRecBkg"), hfHelper.invMassLcToPiKP(candidate), candidate.pt(), efficiencyWeight);
+          registry.fill(HIST("hMassLcVsPtMcRec"), hfHelper.invMassLcToPiKP(candidate), candidate.pt(), efficiencyWeight);
+          registry.fill(HIST("hSelectionStatusMcRec"), candidate.isSelLcToPiKP());
+        }
+      }
+      registry.fill(HIST("hLcPoolBin"), poolBin);
+
+      // Lc-Hadron correlation dedicated section
+      // if the candidate is selected as Lc, search for Hadron ad evaluate correlations
+      for (const auto& track : tracks) {
+        if (std::abs(track.eta()) > etaTrackMax) {
+          continue;
+        }
+        if (track.pt() < ptTrackMin) {
+          continue;
+        }
+        if (std::abs(track.dcaXY()) >= dcaXYTrackMax || std::abs(track.dcaZ()) >= dcaZTrackMax) {
+          continue; // Remove secondary tracks
+        }
+        // Removing Lc daughters by checking track indices
+        if ((candidate.prong0Id() == track.globalIndex()) || (candidate.prong1Id() == track.globalIndex()) || (candidate.prong2Id() == track.globalIndex())) {
+          continue;
+        }
+        registry.fill(HIST("hPtParticleAssocMcRec"), track.pt());
+
+        if (candidate.isSelLcToPKPi() >= selectionFlagLc) {
+          entryLcHadronPair(getDeltaPhi(track.phi(), candidate.phi()),
+                            track.eta() - candidate.eta(),
+                            candidate.pt(),
+                            track.pt(),
+                            poolBin);
+          entryLcHadronRecoInfo(hfHelper.invMassLcToPKPi(candidate), isLcSignal);
+        }
+        if (candidate.isSelLcToPiKP() >= selectionFlagLc) {
+          entryLcHadronPair(getDeltaPhi(track.phi(), candidate.phi()),
+                            track.eta() - candidate.eta(),
+                            candidate.pt(),
+                            track.pt(),
+                            poolBin);
+          entryLcHadronRecoInfo(hfHelper.invMassLcToPiKP(candidate), isLcSignal);
+        }
+      } // end inner loop (Tracks)
+    }   // end outer Lc loop
+    registry.fill(HIST("hZvtx"), collision.posZ());
+    registry.fill(HIST("hMultV0M"), collision.multFV0M());
+  }
+  PROCESS_SWITCH(HfCorrelatorLcHadrons, processMcRec, "Process Mc Reco mode", false);
+
+  /// Lc-Hadron correlation pair builder - for Mc gen-level analysis
+  void processMcGen(aod::McCollision const& mcCollision,
```

Comment:
What's the reason to keep MC gen and MC reco separated in two process functions? Do you need only one of the two in some cases?

Reply:
Ciao Fabrizio, that is because, as also for the HFC D-h correlation codes, the McRec and McGen processes produce and fill the same tables (in particular aod::LcHadronPair), but with different information (gen or rec), so only one of them can be executed in the workflow